### PR TITLE
[doc] new sphnix plugin 

### PR DESCRIPTION
**Pull request by mli**
Updated our doc build pipeline

1. moved functions in sphnix_utils into a sphnix plugin located in mxdoc.py
2. be able to convert markdown files into jupyter notebooks if placed 

    `<!-- INSERT SOURCE DOWNLOAD BUTTONS -->` on some where

    a demo is at http://ec2-52-88-111-46.us-west-2.compute.amazonaws.com/tutorials/basic/ndarray.html
3. removed the dependency to the mxnet-notebooks repo, instead i converted several notebooks into markdowns and placed at docs/
4. mac users can use `tests/ci_build/ci_build.sh doc make -C docs/ html` now if docker is installed.

### CODE REVIEW

File: docs/architecture/overview.md [hunk 1]
```diff
@@ -715,98 +715,6 @@ The gradient, which can be found in `src/operator/smooth_l1_unary-inl.h`, is sim
 The new unified API is designed to fulfill the fundamentals of an operation. For operators with more than two inputs,
 more than one output, or that need more features, see the original [Operator API](http://mxnet.io/architecture/overview.html#operators-in-mxnet).
 
-## KVStore: Multiple Devices and Multiple Computers
```
Comment (piiswrong) [inline docs/architecture/overview.md]: why delete all this?

Comment (mli) [inline docs/architecture/overview.md]: out-of-date and duplicated with another doc

**Comment by mli** [comment]
btw, not ready to merge yet, will update the .md files tonight. also need to test if it works on our doc server

**Comment by mli** [comment]
tested on both linux and mac, should be ok to merge. but not all notebooks converted from mxnet-notebooks are carefully examined. we need to either fix them after merged, or fix them first before merging, which means working on the `doc4` branch for several days.

**Pull request merged**